main.tf

```diff
@@ -66,7 +72,7 @@ resource "google_dns_managed_zone" "forwarding" {
       for_each = var.target_name_server_addresses
       content {
         ipv4_address    = target_name_servers.value.ipv4_address
-        forwarding_path = target_name_servers.value.forwarding_path
+        forwarding_path = try(target_name_servers.value.forwarding_path, null)
```

Comment:
Since `forwarding_path` has [two possible values](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dns_managed_zone#forwarding_path), `default` (This is also default value when the value is not provided for the variable) and `private` you can use following code:

```
lookup(target_name_servers.value, "forwarding_path", "default")
```